runtime/common/src/main/kotlin/kotlinx/serialization/schema/Visitors.kt

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2018 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kotlinx.serialization.schema
+
+import kotlinx.serialization.*
+
+/**
+ * Base interface for a visitor of a [SerialDescriptor] tree,
+ * which can return the result of processing the tree.
+ *
+ * Tree is formed from descriptors using [SerialDescriptor.getElementDescriptor] method.
+ */
+public interface DescriptorVisitor<R> {
```

Comment:
Again, I'd rather move it to the samples.
We can always make it a core functionality as soon as there is any demand on that, but it is better not to publicly expose any API beyond a demand.

Reply:
OK